include/glow/Base/Tensor.h

```diff
@@ -658,6 +658,62 @@ void dumpImpl(const Tensor *T, llvm::raw_ostream &os,
 void dumpImpl(const Tensor *T, unsigned maxNumElem);
 void dumpImpl(const Tensor *T);
 
+template <class ElemTy> class Handle;
+
+template <class ElemTy, bool IsConst>
+class HandleIterator : public std::iterator<std::forward_iterator_tag, ElemTy> {
+  using HandleTy = typename std::conditional_t<IsConst, const Handle<ElemTy> *,
+                                               Handle<ElemTy> *>;
+  using ElemTyRef =
+      typename std::conditional_t<IsConst, const ElemTy &, ElemTy &>;
+
+  HandleTy handle_;
+  llvm::ArrayRef<size_t> sizes_;
+  size_t indices_[max_tensor_dimensions] = {0};
+
+  HandleIterator(HandleTy handle) : handle_(handle) { sizes_ = handle->dims(); }
+
+public:
+  static HandleIterator begin(HandleTy handle) {
```

Comment:
May be Handle class can be a friend class for HandleIterator and only Handle class can create an iterator instance?